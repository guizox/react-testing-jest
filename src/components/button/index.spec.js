import React from "react";
import ReactDom from "react-dom";
import Button from "./index";
import { render, cleanup } from "@testing-library/react";

import renderer from "react-test-renderer";

beforeEach(() => {
  cleanup();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDom.render(
    <Button onClick={() => console.log("click")} label="teste" />,
    div
  );
});

it("renders button correctly with teste", () => {
  const { getByTestId } = render(
    <Button onClick={() => console.log("click")} label="teste" />
  );
  expect(getByTestId("button")).toHaveTextContent("teste");
});

it("renders button correctly with Save", () => {
  const { getByTestId } = render(
    <Button onClick={() => console.log("click")} label="Save" />
  );
  expect(getByTestId("button")).toHaveTextContent("Save");
});

it("matches snapshot 1", () => {
  const tree = renderer
    .create(<Button onClick={() => console.log("click")} label="Save" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer
    .create(
      <Button onClick={() => console.log("click")} label="clcik me please" />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
