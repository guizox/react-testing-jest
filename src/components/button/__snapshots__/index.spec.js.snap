// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<button
  className="button-style"
  data-testid="button"
  onClick={[Function]}
>
  HI 
  Save
</button>
`;

exports[`matches snapshot 2 1`] = `
<button
  className="button-style"
  data-testid="button"
  onClick={[Function]}
>
  HI 
  clcik me please
</button>
`;
